README.md

```diff
@@ -1 +1,160 @@
-# Rotations
+# Rotations.jl
+
+This package implements various rotation parameterizations and defines conversions between them.  All rotation variables are stored as as immutable types.
+
+This package assumes [active (right handed) rotations](https://en.wikipedia.org/wiki/Active_and_passive_transformation) wheere applicable.
+
+### Rotation Parameterizations
+
+1. **Rotation Matrix** `RotMatrix{T <: AbstractFloat}`
+
+    A 3 x 3 rotation matrix storing the rotation.  This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `Mat{3,3,T}`.  A rotation matrix `R` should have the property `I = R * R<sup>T</sup>`.
+
+
+
+2. **Arbitrary Axis Rotation** `AngleAxis{T <: AbstractFloat}`
+
+    A 4 element immutable array with fields `theta`, `x`, `y`, and `z` to store the rotation angle and axis of the rotation.  This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `FixedVectorNoTuple{4, T}`.
+
+
+
+3. **EulerAngles** `EulerAngles{Order <: TaitByranOrder, T <: AbstractFloat}`
+
+    A 3 element immutable array which stores the Euler angles with [**Tait Byran**](https://en.wikipedia.org/wiki/Euler_angles#Tait.E2.80.93Bryan_angles) angle ordering (e.g. `EulerXYZ`) encoded by the template parameter `Order`.   This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `FixedVectorNoTuple{3, T}`.
+
+
+
+4. **ProperEulerAngles** `ProperEulerAngles{Order <: ProperEulerOrder, T <: AbstractFloat}`
+
+    A 3 element immutable array which stores the Euler angles with [**Proper Euler**](https://en.wikipedia.org/wiki/Euler_angles#Conventions) angle ordering (e.g. `EulerXYX`) encoded by the template paramter `Order`.   This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `FixedVectorNoTuple{3, T}`.
+
+
+
+5. **Quaternions** `Quaternion{T <: Real}`
+
+    A 4 element immutable array containing the Quaternion representation of the rotation.  This uses the [Quaternions](https://github.com/JuliaGeometry/Quaternions.jl) package.
+
+
+
+6. **Stereographic Quaternion Projection** `SpQuat{T <: AbstractFloat}`
+
+    A 3 element immutable array containing the stereographic projection of a unit Quaternion.  This gives a compact three element representation of the rotation, but unlike EulerAngles the derivitives of the rotation matrix w.r.t. the `SpQuat` parameters are rational functions.  This makes the `SpQuat` type a good choice for use in optimization problems.
+
+    This projection can be visualized as a pin hole camera, with the pin hole matching the Quaternion `[-1,0,0,0]` and the image plane containing the origin and having normal direction `[1,0,0,0]`.  The "no rotation" `Quaternion(1.0,0,0,0)` then maps to the `SpQuat(0,0,0)`
+
+
+
+
+### Example Usage
+
+```julia
+
+    # create a rotation matrix
+    R = RotMatrix(eye(3))
```

Comment:
Fixed

On 9 May 2016 at 13:12, Chris Foster <notifications@github.com> wrote:

> In README.md
> <https://github.com/FugroRoames/Rotations.jl/pull/2#discussion_r62446724>:
>
> > +
> > +6. **Stereographic Quaternion Projection** `SpQuat{T <: AbstractFloat}`
> > +
> > +    A 3 element immutable array containing the stereographic projection of a unit Quaternion.  This gives a compact three element representation of the rotation, but unlike EulerAngles the derivitives of the rotation matrix w.r.t. the `SpQuat` parameters are rational functions.  This makes the `SpQuat` type a good choice for use in optimization problems.
> > +
> > +    This projection can be visualized as a pin hole camera, with the pin hole matching the Quaternion `[-1,0,0,0]` and the image plane containing the origin and having normal direction `[1,0,0,0]`.  The "no rotation" `Quaternion(1.0,0,0,0)` then maps to the `SpQuat(0,0,0)`
> > +
> > +
> > +
> > +
> > +### Example Usage
> > +
> > +```julia
> > +
> > +    # create a rotation matrix
> > +    R = RotMatrix(eye(3))
>
> This should be eye(RotMatrix) to make the most of FSA efficiencies
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/FugroRoames/Rotations.jl/pull/2/files/fb3a7882257b93afed97cfaa9bf56669bbf410dc#r62446724>
>